### PR TITLE
include provenance properties as image metadata

### PR DESCRIPTION
**Pull request by zhenz**

### CODE REVIEW

File: cc-snapshot [hunk 1]
```diff
@@ -117,6 +118,18 @@ if [ "$(version "$openstackclient_version")" -lt "$(version "3.16.0")" ]; then
 fi
 ###### Check openstackclient version end ######
 
+###### Get provenance properties ######
+PROVENANCE_PROPERTIES=''
+if [ -f "$PROVENANCE_FILE" ]; then
+  for prop in $(cat $PROVENANCE_FILE | tr -d '\n' | tr -d [:space:] | tr -d '{' | tr -d '}' | tr ',' '\n');
```
Comment (diurnalist) [inline cc-snapshot:124]: Can we not assume `jq` is installed? I am not sure if that is one of the base packages we include or not. This solution works fine though!

File: cc-snapshot [hunk 2]
```diff
@@ -117,6 +118,18 @@ if [ "$(version "$openstackclient_version")" -lt "$(version "3.16.0")" ]; then
...
+  do
+    prop="${prop/\":\"/=}"
+    prop="$(echo $prop | tr -d '"')"
+    PROVENANCE_PROPERTIES="--property provenance-$prop $PROVENANCE_PROPERTIES"
```
Comment (diurnalist) [inline cc-snapshot:128]: FYI another way of doing this is to use a shell array

```shell
declare -a PROVENANCE_PROPERTIES=()
PROVENANCE_PROPERTIES+=(--property provenance-$prop)
openstack image create ${PROVENANCE_PROPERTIES[@]} ...
```

**Review by diurnalist — APPROVED** [review]
Looks great!

File: cc-snapshot (hunk 1 above)
Comment (zhenz) [reply cc-snapshot:124]: I don't think we can assume `jq` is installed, and we have https://github.com/ChameleonCloud/cc-snapshot/blob/9092a719adf4570c2b5b993692a910b3cef2d054/cc-snapshot#L195-L211 for parsing vendor data. Actually, I checked and CC-CentOS7 has `jq` installed, but not for both Ubuntu images.

File: cc-snapshot (hunk 2 above)
Comment (zhenz) [reply cc-snapshot:128]: It's good to know! But seems like the bash version of our images couldn't handle empty shell array, and need extra `if` statement. `${PROVENANCE_PROPERTIES[@]}` gives `unbound variable` error if empty.

**Pull request merged**